### DISCUSSION TITLE
a-shell distro ?

### DISCUSSION THREAD

**Discussion by Gbrlueenthunder** [holzschu/a-shell#792] [Q&A]
Hello ,I was trying to 'neofetch' however, it couldn’t install it , why? Well, it said that it couldn’t find a version that satisfies the requirements for neofetch & there is no matching distro found for it ,so is it based off an existing distro (like iSH) OR is it a new one?
(UPDATE : maybe it’s FreeBSD , because the share the same core (1) & kernel (2)(aka Darwin (1) & XNU(2)))

**Comment by holzschu** [comment]
a-Shell is based on the idea of exposing the underlying FreeBSD kernel of iOS, as much as possible. That makes it deeply integrated with the underlying system, including access to the Files app and Shortcuts. If you type `ls /`,  for example, you are seeing the root filesystem of your own phone (a lot of directories are forbidden, and you cannot change anything, but that's still the filesystem of your phone). That means that most of the commands are running natively and using iOS system calls, making them fast. If you type "pickFolder", you can access the documents of many apps on your phone.

On the other hand, that means that installing new commands is not automatic: they either have to be compiled to the native Arm64 system (but then they can only be distributed with the app, as it is not possible to add binaries to an existing app) or to WebAssembly (and then they can be installed with "pkg", but WebAssembly is limited and few commands can be compiled to WebAssembly).

To make matters worse, the FSF has decreed that the GPL is incompatible with AppStore distribution: apps distributed on the AppStore can not be linked with binaries covered by the GPL. I respect this interdiction; the only GPL commands included in a-Shell are compiled to WebAssembly, and thus not part of the app binary.

Short answer: a-Shell is using its own distribution of packages. neofetch requires bash, which is GPL and thus not available on a-Shell, and also makes a lot of system calls that are not available on iOS.
